### PR TITLE
Source of truth

### PR DESCRIPTION
**Pull request by pYr0x**
this PR should resolve the problems with steal-tools. all tests in steal-tools should now pass with this PR

i tested steal-tools with this commit of my fork https://github.com/stealjs/steal-tools/commit/c18212b343b0554c1f55ffc2e720d5187d8ad635
travis pass: https://travis-ci.org/stealjs/steal-tools/builds/121008438

@matthewp but i dont know why there is a conflict with merging, i tried very thing but without success. can you please look into the problem...

the main changes are:
* remove `if(this.stealBundled && this.stealBundled === true) {` because `if(!this.loadBundles) {` does it better
* update the `loadBundles` setter

### CODE REVIEW

**Comment by matthewp** [comment]
What issues is this supposed to fix again?

**Comment by pYr0x** [comment]
this fix this [issue](https://github.com/stealjs/steal/issues/631) and your test yesterday to check steal-tools against master of steal https://github.com/stealjs/steal-tools/tree/fail

**Comment by matthewp** [comment]
So what was going on there? This makes the url options the source of truth, so was the `steal = {}` options overriding the url options? If so, which ones?

**Comment by pYr0x** [comment]
so very simple.
if a config is provided e.g. `steal = {}` we extend the config with the `urlOptions`
```
if(typeof config === 'object'){
			// the url options are the source of truth
			config = extend(config, urlOptions);
		}else{
			config = urlOptions;
		}
```

**Comment by matthewp** [comment]
Would you mind adding a test with the breaking scenario? I still don't have it clearly in my head. That would help me review this better.

**Comment by pYr0x** [comment]
there is a test included `test/stealconfig/dev.html`

**Comment by matthewp** [comment]
But this PR doesn't include any changes to that test. It was passing before these changes, no?

**Comment by pYr0x** [comment]
The test will not passing without the changes...

**Comment by matthewp** [comment]
So this is a new test that's not in master? It's weird that github is not showing it as being a new test...

**Comment by matthewp** [comment]
Yeah, weird, this commit has the test https://github.com/stealjs/steal/pull/655/commits/3a28d41eb271462306d7fa8226edd4c7ce123b78

File: src/config.js [hunk 1]
```diff
@@ -228,17 +232,15 @@
 					stealPath = paths.join("/"),
 					platform = this.getPlatform() || (isWebWorker ? "worker" : "window");
 
-				// if steal is bundled we always are in production environment
-				if(this.stealBundled && this.stealBundled === true) {
-					this.config({ env: platform+"-production" });
-
-				}else{
+				// we dont want to set all the development paths if loadBundles is true
```
Comment (matthewp) [inline src/config.js]: Why not, what problem does it cause?

**Comment by pYr0x** [comment]
@matthewp maybe we have to talk about that, it is hard to explain what i have done...
we can look into this in the next hangout

**Comment by pYr0x** [comment]
close this for https://github.com/stealjs/steal/pull/757

**Pull request closed without merging**